src/core/server/saved_objects/migrations/core/migration_context.ts

```diff
@@ -199,6 +208,37 @@ export function disableUnknownTypeMappingFields(
   };
 }
 
+/**
+ * This function is used to modify the target mappings object by deleting specified type mappings fields.
+ *
+ * The function operates under the following conditions:
+ * - It checks if the 'migrations.delete.enabled' configuration is set to true.
+ * - If true, it retrieves the 'migrations.delete.types' configuration
+ * - For each type, it deletes the corresponding property from the target mappings object.
+ *
+ * The purpose of this function is to allow for dynamic modification of the target mappings object
+ * based on the application's configuration. This can be useful in scenarios where certain type
+ * mappings are no longer needed and should be removed from the target mappings.
+ *
+ * Note: The function does not return a value. It directly modifies the targetMappings object passed to it.
```

Comment:
i really like the clear code and jsdoc for the method. One question about mutable input param, it is generally safer to return a new value from a function instead of modifying its inputs in functional programming, what's the rational behind to modify the targetMappings?

Reply:
refer to https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#avoid-mutability-and-state

Reply:
> generally safer to return a new value from a function instead of modifying its inputs in functional programming

yeah. i originally had the spread operation as well and noticed the code became even harder to understand what is happening here.

i can update to mimic the other behavior when marking types unknown

Reply:
updated